packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -152,24 +157,30 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
 
         uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
 
-        for (uint i = 0; i < candidates.length; i++) {
-            address candidate = candidates[i];
+        address[] memory candidates = new address[](sortedCandidates.length);
```

Comment:
Maybe name this `prioritizedCandidates` and comment a bit about the algo?
Something like: "The goal is to merge the two arrays from the interface, resulting in a single array with the candidates in priority order. The interface requests the info in two arrays in order to guarantee no duplicates with lower computational cost."

Btw, leaving a lot of comments as mini improvement suggestions, but I love the algorithm :)

Reply:
Done! https://github.com/Synthetixio/synthetix-v3/pull/572/files#diff-69650d386e562b4dac992709531657fe63fedd176cde00f236b7b6271333c70eR173